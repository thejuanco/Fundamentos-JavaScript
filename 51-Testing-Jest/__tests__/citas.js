import Citas from "../js/classes/Citas";

describe("Probar la clase de citas", () => {
    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Rex',
            propietario: 'Juan',
            telefono: '1234567890',
            fecha: '2022-01-15',
            hora: '10:00',
            sintomas: 'Sangre fría'
        };
        citas.agregarCita(citaObj);

        //Prueba
        expect(citas).toMatchSnapshot();
    })

    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Hook',
            propietario: 'Juan M.',
            telefono: '1234567890',
            fecha: '2022-01-15',
            hora: '10:00',
            sintomas: 'Sangre fría'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    })

    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        //Pruebas
        expect(citas).toMatchSnapshot()
    });
})