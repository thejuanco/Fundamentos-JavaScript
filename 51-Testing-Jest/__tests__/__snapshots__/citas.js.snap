// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "2022-01-15",
      "hora": "10:00",
      "id": 1729570731632,
      "mascota": "Hook",
      "propietario": "Juan M.",
      "sintomas": "Sangre fría",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "2022-01-15",
      "hora": "10:00",
      "id": 1729570731632,
      "mascota": "Rex",
      "propietario": "Juan",
      "sintomas": "Sangre fría",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
